### PR TITLE
feat(dashboard): remove from codesandbox

### PR DESCRIPTION
**Pull request by olarclara**
## What kind of change does this PR introduce?

Allows users to remove branches and repositories from CodeSandbox through the dashboard interface.

|Repository|Branch|
|:-:|:-:|
|Screen Shot 2022-10-10 at 16 36 36|Screen Shot 2022-10-10 at 16 36 56|

Repositories are unlinked from the current team.
Branches are removed from the repository but not deleted on GitHub.

## What is the new behavior?

Demo: https://www.loom.com/share/51ec5eb3db2d4fcd9458940e51418a7f

### CODE REVIEW

**Comment by lbogdan** [comment]
Build for latest commit https://github.com/codesandbox/codesandbox-client/pull/7012/commits/67230fd18ecaecf300684494111b889caf73f1dd is at https://pr7012.build.csb.dev/s/new.

File: packages/app/src/app/pages/Dashboard/Components/Selection/ContextMenus/BranchMenu.tsx [hunk 1]
```diff
@@ -56,15 +61,23 @@ export const BranchMenu: React.FC<BranchMenuProps> = ({ branch }) => {
       <MenuItem onSelect={() => history.push(repoUrl)}>
         Open repository
       </MenuItem>
-      {/* TODO: Implement remove branch <Menu.Divider />
       <Menu.Divider />
+      {/** TODO: disabled state */}
```
Comment (tristandubbeld) [inline packages/app/src/app/pages/Dashboard/Components/Selection/ContextMenus/BranchMenu.tsx:65]:
```suggestion
```

If this is about the `MenuItem` below :)

**Review by alexnm — APPROVED** [review]
Great work! Left some comments, but giving the approval as I don't think a second pass is needed

File: packages/app/src/app/overmind/effects/api/index.ts [hunk 2]
```diff
@@ -691,4 +691,10 @@ export default {
       `/teams/${teamId}/customer_portal?return_path=${return_path}`
     );
   },
+  removeBranchFromRepository(owner: string, repo: string, branch: string) {
+    return api.delete(`/beta/sandboxes/github/${owner}/${repo}/${branch}`);
```
[the remark this answers is not in the document]
Comment (olarclara) [reply packages/app/src/app/overmind/effects/api/index.ts:695]: @tristandubbeld nope, the API's archiving branches feature is under the beta controller. The action is implemented here: https://github.com/codesandbox/codesandbox-client/blob/master/packages/app/src/app/overmind/effects/api/index.ts#L513.

File: packages/app/src/app/overmind/namespaces/dashboard/actions.ts [hunk 3]
```diff
@@ -2035,3 +2045,91 @@ export const getRepositoryByDetails = async (
...
+            repo.repository.owner === owner &&
+            repo.repository.name === repoName
+          ) {
+            repo.branches = repo.branches.filter(b => b.id !== id);
+            acc.push(repo);
+          } else {
+            acc.push(repo);
+          }
+          return acc;
+        }, [] as Repository[]) ?? [];
```
[the remark this answers is not in the document]
Comment (olarclara) [reply packages/app/src/app/overmind/namespaces/dashboard/actions.ts]:
```ts
      const repository = state.dashboard.repositories?.find(
        r => r.repository.owner === owner && r.repository.name === repoName
      );

      if (repository) {
        // Manually remove the data from the state.
        repository.branches = repository.branches.filter(b => b.id !== id);
      }
```

https://github.com/codesandbox/codesandbox-client/pull/7012/commits/268ddf0c7559c3d5bd859ae40094d6f5f7264751#diff-e77b3a86aa6cab847b5bda8da7259d02353b22b91171ae6d506f6bc6aa65c282R2115-R2122

File: packages/app/src/app/overmind/namespaces/dashboard/actions.ts [hunk 4]
```diff
@@ -1968,10 +1969,19 @@ export const getContributionBranches = async ({ state, effects }: Context) => {
...
+};
+export const getRepositoriesByTeam = async (
+  { state, effects }: Context,
+  options?: RepositoriesActionOptions
+) => {
   const { activeTeam, dashboard } = state;
+  const { bypassLoading = false } = options ?? {};
   try {
-    dashboard.repositories = null;
+    if (!bypassLoading) {
```
[the remark this answers is not in the document]
Comment (olarclara) [reply packages/app/src/app/overmind/namespaces/dashboard/actions.ts:1985]: Makes sense! Updated the implementation and added an explanation for not calling `syncData` if the loading state is bypassed. https://github.com/codesandbox/codesandbox-client/pull/7012/commits/268ddf0c7559c3d5bd859ae40094d6f5f7264751#diff-e77b3a86aa6cab847b5bda8da7259d02353b22b91171ae6d506f6bc6aa65c282R1981-R1983

File: packages/app/src/app/overmind/effects/api/index.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (tristandubbeld) [reply packages/app/src/app/overmind/effects/api/index.ts:695]: Cool, thanks for answering! :)

**Pull request merged**